### PR TITLE
Remove unnecessary reflection code from JavaTemplates

### PR DESCRIPTION
**Pull request by Shredder121**

### CODE REVIEW

File: querydsl-core/src/test/java/com/querydsl/core/serialization/JavaTemplatesTest.java [hunk 1]
```diff
@@ -16,27 +16,21 @@
...
+        Templates templates = new JavaTemplates();
         int matched = 0;
-        for (Field field : Ops.class.getFields()) {
-            if (Operator.class.isAssignableFrom(field.getType())) {
-                matched++;
-                Operator operator = (Operator) field.get(null);
-                assertNotNull(field.getName() + " missing", templates.getTemplate(operator));
-            }
+        for (Operator operator : Ops.values()) {
+            ++matched;
```
Comment (Shredder121) [inline querydsl-core/src/test/java/com/querydsl/core/serialization/JavaTemplatesTest.java]: This doesn't do anything anymore.
you could assert that the `Ops.values()` is not empty :smile:

**Comment by johnktims** [comment]
I contemplated removing it and just decided to leave it. Ops is never going to be empty so I think I can safely remove that final assertion altogether.

File: querydsl-core/src/test/java/com/querydsl/core/serialization/JavaTemplatesTest.java (hunk 1 above)
Comment (johnktims) [inline querydsl-core/src/test/java/com/querydsl/core/serialization/JavaTemplatesTest.java]: I contemplated removing it and just decided to leave it. Ops is never going to be empty so I think I can safely remove that final assertion altogether.

**Comment by Shredder121** [comment]
I am rebasing this, fixup for [imports of 871b998](https://github.com/querydsl/querydsl/blob/871b998ed4c1fa08f7a7b2f14533a72191e92878/querydsl-core/src/main/java/com/querydsl/core/types/JavaTemplates.java#L16).

**Pull request merged**